### DISCUSSION TITLE
If the paths defined are relative to URL defined under server tag, the stubs are not getting recognized and loaded during tests.

### DISCUSSION THREAD

**Discussion by prateekkarmakar** [specmatic/specmatic#1470] [Ask For Help] [answered]
### I have an order.yaml with the following :

openapi: 3.0.1
info:
  title: OpenAPI definition
  version: v0
servers:
  - url: http://localhost:9000/orders/customer
    description: Generated server url
....
paths:
    /{customerId}:
    get:
      tags:
        - order-controller
      operationId: fetchOrdersForCustomer
      parameters:
        - name: customerId
          in: path
          required: true
          schema:
            type: integer
            format: int32
      responses:
        '200':
          description: OK
          content:
            'application/json':
              schema:
                type: array
                items:
                  $ref: '#/components/schemas/CustomerOrder'

### When I define a stub with the following approach in an external file. The following stub is not picked up and while running the te

{
  "http-request": {
    "method": "GET",
    "path": "/orders/customer/1001"
  },
  "http-response": {
    "status": 200,
    "body": [{
      "id": 1,
      "orderId": "ORD001",
      "customerId": 1001,
      "placeDate": "2022-01-01",
      "completionDate": "2022-01-02"
    }]
  }
}

### When the test run starts the stub files are not picked up with following error

API Specification Summary: .\local-spec\order.yaml
  OpenAPI Version: 3.0.1
  API Paths: 2, API Operations: 2
  Schema components: 1, Security Schemes: none

  Loading stub expectations from src\test\resources\order_examples
  Reading the following stub files:
    src\test\resources\order_examples\order1.json
    src\test\resources\order_examples\order2.json
  src\test\resources\order_examples\order1.json didn't match any of the contracts
    No matching REST stub or contract found for method GET and path /orders/customer/1001 (assuming you're looking for a REST API since no SOAPAction header was detected)
  src\test\resources\order_examples\order2.json didn't match any of the contracts
    No matching REST stub or contract found for method GET and path /orders/customer/1002 (assuming you're looking for a REST API since no SOAPAction header was detected) 

### Also while running the tests, I get the following error

No matching REST stub or contract found for method GET and path /orders/customer/1001 (assuming you're looking for a REST API since no SOAPAction header was detected)

### The Stubs are picked up when I change the path as follows:

openapi: 3.0.1
info:
  title: OpenAPI definition
  version: v0
servers:
  - url: http://localhost:9000
    description: Generated server url
....
paths:
    /orders/customer/{customerId}:

**As per the above I find the relative paths are not considered while picking up the stubs. Suppose the user has no way of changing the contract and has to rely on externalizing the stubs. How can the user configure the stubs so that it gets picked up during the run?**

I have attached the order.yaml (as order.txt) with relative paths and the logs of the two runs and sample stub. for reference
[order1.json](https://github.com/user-attachments/files/18111625/order1.json) 
[failure_logs.txt](https://github.com/user-attachments/files/18111533/failure_logs.txt)
[success_logs.txt](https://github.com/user-attachments/files/18111535/success_logs.txt)
[order.txt](https://github.com/user-attachments/files/18111553/order.txt)

**Comment by harikrishnan83** [comment, marked as answer]
hello @prateekkarmakar , thanks for reaching out.

The problem where we cannot change the spec to add path prefixes (to simulate gateways, etc.), add headers, etc. is quite common. The best approach IMHO for this is to use [OpenAPI Overlays](https://spec.openapis.org/overlay/v1.0.0.html) to add these changes externally.

Specmatic supports overlays and here is the [documentation](https://github.com/znsio/specmatic/discussions/1495#discussioncomment-11656450). In your specific case, we can add the `/orders/customer` prefix to your path externally through an overlay file and Specmatic will apply this change on your OpenAPI spec at runtime and there by pick up your example JSONs (which can continue have `/orders/customer/`). Hope this helps.

Please reach out to us if you need any help on this.

Regards,
HariKrishnan

**Comment by prateekkarmakar** [comment]
@harikrishnan83, Can You share an example for the above I have tried some things but its not working neither in the OpenAPI documentation there is any reference.

I tried things like this

overlay: 1.0.0
info:
  title: Append /orders to Paths
  version: 0.0.1
actions:
  - target: "$.paths['/mobile/customer']"
    update:
      paths: "/orders/mobile/customer"

but I get the following error

The OpenAPI file .\.\local-specs\order.yaml was read successfully but with some issues
Error parsing file .\.\local-specs\order.yaml
  attribute paths.'/mobile/customer'.paths is unexpected
